apps/src/music/views/Keybed.tsx

```diff
@@ -73,8 +96,43 @@ const Key: React.FunctionComponent<KeyProps> = ({
   text,
   isVertical,
 }: KeyProps) => {
+  const handleKeyDown = (event: React.KeyboardEvent<HTMLDivElement>) => {
+    const keyElements = Array.from(
+      document.querySelectorAll<HTMLElement>(`#${keyId}`)
+    );
+    const currentIndex = keyElements.indexOf(event.currentTarget);
+    if (event.key === 'ArrowRight') {
+      // Move focus to the next sibling
+      const nextIndex = (currentIndex + 1) % keyElements.length;
+      keyElements[nextIndex].focus();
+    } else if (event.key === 'ArrowLeft') {
+      // Move focus to the previous sibling
+      const previousIndex =
+        (currentIndex - 1 + keyElements.length) % keyElements.length;
+      keyElements[previousIndex].focus();
+    } else if (event.key === 'Enter' || event.key === ' ') {
+      // Trigger the onClick handler for selection
+      onClick();
+      event.preventDefault();
+      // Stop propagation to prevent enter setting focus on first element
+      event.stopPropagation();
+    } else if (event.key === 'Escape') {
+      // Stop propagation to prevent escape from closing the entire popup
+      event.stopPropagation();
+      // Move focus back to the keyPad container
+      const keypadContainer = document.getElementById('keypad');
+      if (keypadContainer) {
+        keypadContainer.focus();
+      }
+    } else if (event.key === 'Tab') {
+      // Swallow the event so the focus doesn't leave the keybed
+      event.stopPropagation();
+    }
+  };
+
   return (
     <div
+      id={keyId}
```

Comment:
Technically speaking, should the ID of any given element always be unique? Would it be better to target these using a class name or `data-` attribute?

Reply:
Yes definitely! Is it alright if I clean up both in their own followup PR?

Reply:
Works for me!